src/box/gc.c

```diff
@@ -903,14 +903,17 @@ gc_erase_consumer_on_commit(struct trigger *trigger, void *event)
 int
 gc_erase_consumer(const struct tt_uuid *uuid)
 {
-	assert(in_txn() != NULL);
 	struct gc_consumer *consumer = gc_consumer_by_uuid(uuid);
 	if (!gc_schema_supports_persistent_consumers()) {
 		if (consumer != NULL)
 			consumer->is_persistent = false;
 		return 0;
 	}
 
+	bool is_autocommit = in_txn() == NULL;
+	if (is_autocommit && txn_begin() == NULL)
+		return -1;
+
 	if (boxk(IPROTO_DELETE, BOX_GC_CONSUMERS_ID, "[%s]",
 		 tt_uuid_str(uuid)) != 0)
 		return -1;
```

Comment:
Need to roll the autocommit txn back on failure. Otherwise the transaction will stay in the fiber and won't allow to start any new transaction.

Reply:
Oh, thanks!
Fixed and covered with a test case.